### PR TITLE
Add PKCE and State to Auth Code Exchange

### PR DESCRIPTION
**Pull request by d4nleonard**
Adding PKCE and state to the authorization code exchange. Also adding files to launch with VS Code instead of MAMP

### CODE REVIEW

**Comment by Blackbaud-ChristopherRodgers** [comment]
Changes look good. Might need to make a note about the PHP 7+ requirement. Thanks for leveling up the auth logic (taking token expiration into account). Tutorial likely needs updates for the code changes here. We also need to note that setting the application Scopes is a requirement to run this sample. Might also help if the UI presented the error on failure.

Scopes failure is returned as a 200 response by the endpoint currently.
```
{
  "constituent": {
    "title":"Forbidden",
    "status":403,
    "detail":"This application has insufficient scope to perform the operation. The access token used in this request has no data access. This application's scopes configuration is either set to \"No access mode\" or the Blackbaud Environment Admin has not yet approved the application's updated access configuration. Learn more: https:\/\/developer.blackbaud.com\/skyapi\/docs\/applications\/scopes"
  }
}
```
image

**Comment by lindsey-rix** [comment]
Readme changes look good.

File: README.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ A local running Apache server to run and host the project.  For this example we
 
 To run this application in your environment, follow the [Auth Code Flow Tutorial](https://developer.blackbaud.com/skyapi/docs/code/auth-flows/auth-code-flow) instructions within our documentation. 
 
-> PHP sample coming soon.   Follow the instructions for obtaining the necessary keys from the other code samples.
+Note: Make sure your application access is properly set on your SKY application. The application access must be full or limited. If it is limited, the rnxt.r application scope is required.
```
Comment (Blackbaud-ChristopherRodgers) [inline README.md]: We no longer show `rnxt.r` in the My Apps - Scopes UI. We might want to say "Raiser's Edge NXT - Read" instead. Could be helpful to link the Scopes docs here as well.
What do you think, @Blackbaud-LindseyRix ?

Comment (d4nleonard) [reply README.md]: Updated the text

**Review by Blackbaud-ChristopherRodgers — APPROVED** [review]

**Pull request merged**